tests/test_training.py

```diff
@@ -0,0 +1,146 @@
+import os
+import shutil
+
+import pytest
+import sentencepiece as spm
+from fixtures import DataDir, en_sample, ru_sample
+
+current_folder = os.path.dirname(os.path.abspath(__file__))
+fixtures_path = os.path.join(current_folder, "fixtures")
+root_path = os.path.abspath(os.path.join(current_folder, ".."))
+bin_dir = os.environ["BIN"] if os.getenv("BIN") else os.path.join(root_path, "bin")
+marian_dir = (
+    os.environ["MARIAN"]
+    if os.getenv("MARIAN")
+    else os.path.join(root_path, "3rd_party", "marian-dev", "build")
+)
+
+
+def validate_alignments(corpus_path, model_path):
+    sp = spm.SentencePieceProcessor(model_file=model_path)
+
+    with open(corpus_path) as f:
+        for line in f:
+            fields = line.strip().split("\t")
+            assert len(fields) == 3
+            src = sp.encode_as_pieces(fields[0])
+            trg = sp.encode_as_pieces(fields[1])
+            alignment = [[int(num) for num in pair.split("-")] for pair in fields[2].split()]
+
+            for idx_src, idx_trg in alignment:
+                try:
+                    assert src[idx_src] is not None
+                    assert trg[idx_trg] is not None
+                except:
+                    print("src: ", src)
+                    print("trg: ", trg)
+                    print("alignment:", alignment)
+                    raise
+
+
+@pytest.fixture()
+def data_dir():
+    return DataDir("test_training")
+
+
+@pytest.fixture()
+def vocab(data_dir):
+    shutil.copyfile("tests/data/vocab.spm", data_dir.join("vocab.spm"))
+
+
+@pytest.fixture()
+def corpus(data_dir):
+    data_dir.create_zst("corpus.en.zst", en_sample)
+    data_dir.create_zst("corpus.ru.zst", ru_sample)
+    data_dir.create_zst("devset.en.zst", en_sample)
+    data_dir.create_zst("devset.ru.zst", ru_sample)
+
+
+@pytest.fixture()
+def alignments(data_dir, vocab, corpus):
+    env = {
+        "TEST_ARTIFACTS": data_dir.path,
+        "BIN": bin_dir,
+        "MARIAN": marian_dir,
+        "COMPRESSION_CMD": "zstd",
+        "ARTIFACT_EXT": "zst",
+        "SRC": "en",
+        "TRG": "ru",
+    }
+    data_dir.run_task("alignments-en-ru", env=env)
+    shutil.copyfile(
+        os.path.join(data_dir.path, "artifacts", "corpus.aln.zst"), data_dir.join("corpus.aln.zst")
+    )
+
+
+def test_train_student_mocked(alignments, data_dir):
+    """
+    Run training with mocked marian to check OpusTrainer output
```

Comment:
Praise: Thanks for the nice explainer on the assumption for what this test is testing.